### DISCUSSION TITLE
Some questions about enabling deepspeed in cogvideo training scripts

### DISCUSSION THREAD

**Discussion by luyvlei** [huggingface/diffusers#9695] [Q&A]
In this 1448 line of this script, when deepspeed is enabled, optimizer.step() and zero_grad() will no longer be executed. I have checked the related documentation of accelerate and found no similar introduction, is this correct?

https://github.com/huggingface/diffusers/blob/d9029f2c5981a96ab51f8996be620af116b8d743/examples/cogvideo/train_cogvideox_lora.py#L1448-L1450

**Comment by a-r-r-o-w** [comment]
The optimizer step and gradient zero-ing is handled internally within the deepspeed engine I think (not 100% sure), so it is okay to skip. I believe even if executed, these will be no-ops. I think to reduce confusion, we could do these but I don't have the bandwidth to test on a complete run. Would you be able to verify? cc @yiyixuxu
